pycbc/events/stat.py

```diff
@@ -792,6 +1093,102 @@ def coinc_multiifo(self, s, slide, step, to_shift,
+        singles['sigmasq'] = trigs['sigmasq'][:]
+        singles['snr'] = trigs['snr'][:]
+        try:
+            tnum = trigs.template_num  # exists if accessed via coinc_findtrigs
+        except AttributeError:
+            tnum = trigs['template_id']  # exists for SingleDetTriggers
+            # Should only be one ifo fit file provided
+            assert len(self.ifos) == 1
+        # store benchmark log volume as single-ifo information since the coinc
+        # method does not have access to template id
+        singles['benchmark_logvol'] = self.benchmark_logvol[tnum]
+        return numpy.array(singles, ndmin=1)
+
+    def coinc_multiifo(self, s, slide, step, to_shift,
+                       **kwargs): # pylint:disable=unused-argument
+        sngl_rates = {sngl[0]: sngl[1]['snglstat'] for sngl in s}
+        ln_noise_rate = coinc_rate.combination_noise_lograte(
+                                  sngl_rates, kwargs['time_addition'])
+        ln_noise_rate -= self.benchmark_lograte
+
+        # Network sensitivity for a given coinc type is approximately
+        # determined by the least sensitive ifo
+        network_sigmasq = numpy.amin([sngl[1]['sigmasq'] for sngl in s],
+                                     axis=0)
+        # Volume \propto sigma^3 or sigmasq^1.5
+        network_logvol = 1.5 * numpy.log(network_sigmasq)
+        # Get benchmark log volume as single-ifo information
+        # NB benchmark logvol for a given template is not ifo-dependent
+        # - choose the first ifo for convenience
+        benchmark_logvol = s[0][1]['benchmark_logvol']
+        network_logvol -= benchmark_logvol
+
+        stat = {ifo: st for ifo, st in s}
+        logr_s = self.logsignalrate_multiifo(stat,
+                                             slide * step, to_shift)
+
+        loglr = logr_s + network_logvol - ln_noise_rate
+        # cut off underflowing and very small values
+        loglr[loglr < -30.] = -30.
+        return loglr
+
+class TwoGCStatistic(ExpFitSGFgBgRateNewStatistic):
+    def __init__(self, files, ifos=None):
+        ExpFitSGFgBgRateNewStatistic.__init__(self, files, ifos=ifos)
+        self.get_newsnr = ranking.get_newsnr_sgveto_psdvar_scaled
+
+class TwoGCBBHStatistic(ExpFitSGFgBgRateNewStatistic):
+    def __init__(self, files, ifos=None):
+        ExpFitSGFgBgRateNewStatistic.__init__(self, files, ifos=ifos)
+        self.get_newsnr = ranking.get_newsnr_sgveto_psdvar_scaled_threshold
+
+    def single(self, trigs):
+        from pycbc.conversions import mchirp_from_mass1_mass2
+        self.mchirp = mchirp_from_mass1_mass2(trigs.param['mass1'], trigs.param['mass2'])
+        return ExpFitSGFgBgRateNewStatistic.single(self, trigs)
+
+    def logsignalrate_multiifo(self, stats, shift, to_shift):
+        logr_s = ExpFitSGFgBgRateNewStatistic.logsignalrate_multiifo(self, stats, shift, to_shift)
+        logr_s += numpy.log((self.mchirp / 20.0) ** (11./3.0))
```

Comment:
comment to explain? (eg 'model signal rate as uniform over mchirp, background rate propto mchirp^(-11/3) due to bank density')